cd/auto-generate/auto-generate-controllers.sh

```diff
@@ -0,0 +1,241 @@
+fi
+
+# find all the directories whose name ends with 'controller'
+pushd "$WORKSPACE_DIR" >/dev/null
+  CONTROLLER_NAMES=$(find . -maxdepth 1 -mindepth 1 -type d | cut -d"/" -f2 | grep -E "controller$")
+popd >/dev/null
+
+for CONTROLLER_NAME in $CONTROLLER_NAMES; do
+  SERVICE_NAME=$(echo "$CONTROLLER_NAME"| sed 's/-controller$//g')
+  print_line_separation
+
+  echo "auto-generate-controllers.sh][INFO] ## Generating new controller for $SERVICE_NAME service ##"
+  # if the go.mod file is missing in a service controller, skip auto-generation
+  if [[ ! -f "$WORKSPACE_DIR/$CONTROLLER_NAME/go.mod" ]]; then
+    echo "auto-generate-controllers.sh][ERROR] 'go.mod' file is missing. Skipping $CONTROLLER_NAME."
+    continue
+  fi
+
+  # Find the ACK runtime version in service controller 'go.mod' file
+  # If the current version is same as latest ACK runtime version, skip this controller.
+  SERVICE_RUNTIME_VERSION=$(grep "github.com/aws-controllers-k8s/runtime" "$WORKSPACE_DIR/$CONTROLLER_NAME/go.mod" | grep -oE "v[0-9]+\.[0-9]+\.[0-9]+")
+  if [[ $SERVICE_RUNTIME_VERSION == $ACK_RUNTIME_VERSION ]]; then
+    echo "auto-generate-controllers.sh][INFO] $CONTROLLER_NAME already has the latest ACK runtime version $ACK_RUNTIME_VERSION. Skipping $CONTROLLER_NAME."
+    continue
+  fi
+
+  echo "auto-generate-controllers.sh][INFO] ACK runtime version for new controller will be $ACK_RUNTIME_VERSION. Current version is $SERVICE_RUNTIME_VERSION."
+  echo "auto-generate-controllers.sh][INFO] Generating new controller code using command 'make build-controller'."
+  export SERVICE=$SERVICE_NAME
+  MAKE_BUILD_OUTPUT_FILE=/tmp/"$SERVICE_NAME"_make_build_output
+  MAKE_BUILD_ERROR_FILE=/tmp/"$SERVICE_NAME"_make_build_error
+  if ! make build-controller > "$MAKE_BUILD_OUTPUT_FILE" 2>"$MAKE_BUILD_ERROR_FILE"; then
+    cat "$MAKE_BUILD_ERROR_FILE"
+
+    echo "auto-generate-controllers.sh][ERROR] failure while executing 'make build-controller' command. Creating/Updating GitHub issue."
+    ISSUE_TITLE="Errors while generating $CONTROLLER_NAME for ACK runtime $ACK_RUNTIME_VERSION"
+
+    echo -n "auto-generate-controllers.sh][INFO] Querying already open GitHub issue ... "
+    ISSUE_NUMBER=$(gh issue list -R "$GH_ORG/$GH_ISSUE_REPO" -L 1 -s open --json number -S "$ISSUE_TITLE" --jq '.[0].number' -A @me -l "$GH_LABEL")
+    if [[ $? -ne 0 ]]; then
+      echo ""
+      echo "auto-generate-controllers.sh][ERROR] unable to query open github issue. Skipping $CONTROLLER_NAME."
+      continue
+    fi
+    echo "ok."
+
+    # Capture 'make build-controller' command output & error, then persist
+    # in '$GH_ISSUE_BODY_FILE'
+    MAKE_BUILD_OUTPUT=$(cat "$MAKE_BUILD_OUTPUT_FILE")
+    MAKE_BUILD_ERROR_OUTPUT=$(cat "$MAKE_BUILD_ERROR_FILE")
+    GH_ISSUE_BODY_TEMPLATE_FILE="$THIS_DIR/gh_issue_body_template.txt"
+    GH_ISSUE_BODY_FILE=/tmp/"SERVICE_NAME"_gh_issue_body
+    eval "echo \"$(cat "$GH_ISSUE_BODY_TEMPLATE_FILE")\"" > $GH_ISSUE_BODY_FILE
```

Comment:
https://unix.stackexchange.com/a/294840 Maybe?

Reply:
I did look at it but this tool needs to be installed separately and eval-echo works just fine.